### ISSUE TITLE
'readable' event fired only once in IncomingMessage response

### ISSUE THREAD

**Issue by jonathan-roy** [electron/electron#21938]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 7.*
* **Operating System:**
  * macOS
  * Linux
  * not tested on Windows
* **Last Known Working Electron version:**
  * 6.*

### Expected Behavior

When using the `net` module to send an HTTP request, the `IncomingMessage` response should fired the `readable` event until the entire response is read.

### Actual Behavior

The `readable` event is fired only once.

### To Reproduce

```javascript
const {net} = require("electron");

// When the app is ready:

const request = net.request({
    "url": "https://www.google.com",
    "method": "GET",
    "redirect": "follow"
});

request.on("response", (response) => {
    response.on("readable", () => {
            const bufferSize = 4 * 1024;
            const buffer = response.read(bufferSize);
            if (!buffer) {
                    return;
            }
            //console.log(buffer.toString());
            console.log("read");
    });
});

request.end();
```

**Comment by electron-triage** [comment]
The Electron version reported on this issue is no longer supported. See our [supported versions documentation](https://www.electronjs.org/docs/tutorial/support#supported-versions).

If this is still reproducible on a supported version, please open a new issue with any other new information that a maintainer should know.

Thank you for taking the time to report this issue and helping to make Electron better! Your help is appreciated.
